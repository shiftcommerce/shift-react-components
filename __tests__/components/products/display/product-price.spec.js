// Libraries
import React from 'react'

// Component
import ProductPrice from '../../../../src/components/products/display/product-price'

// Fixtures
import ProductWithVariantsWithSimilarPrice from '../../../fixtures/product-with-variants-with-similar-price'
import ProductWithVariantsWithDifferentPrices from '../../../fixtures/product-with-variants-with-different-prices'

test('renders correctly if variants have similar price', () => {
  // Act
  const wrapper = mount(
    <ProductPrice
      minPrice={ProductWithVariantsWithSimilarPrice.min_current_price}
      maxPrice={ProductWithVariantsWithSimilarPrice.max_current_price} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('£10.00')
})

test('renders correctly if variants have different prices', () => {
  // Act
  const wrapper = mount(
    <ProductPrice
      minPrice={ProductWithVariantsWithDifferentPrices.min_current_price}
      maxPrice={ProductWithVariantsWithDifferentPrices.max_current_price} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('£10.00 - £12.00')
})
