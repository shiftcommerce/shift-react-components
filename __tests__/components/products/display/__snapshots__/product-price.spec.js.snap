// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly if variants have different prices 1`] = `
<ProductPrice
  maxPrice="12.00"
  minPrice="10.00"
>
  <div
    className="c-product-price"
  >
    <span>
      £
      10.00
       - £
      12.00
    </span>
  </div>
</ProductPrice>
`;

exports[`renders correctly if variants have similar price 1`] = `
<ProductPrice
  maxPrice="10.00"
  minPrice="10.00"
>
  <div
    className="c-product-price"
  >
    <span>
      £
      10.00
    </span>
  </div>
</ProductPrice>
`;
